src/lib/components/TokenSelect/index.tsx

```diff
@@ -65,55 +65,54 @@ export function TokenSelectDialog({ value, onSelect }: TokenSelectDialogProps) {
     [query, areBalancesLoaded, isTokenListLoaded]
   )
 
-  const baseTokens: Currency[] = [] // TODO(zzmp): Add base tokens to token list functionality
-
   const input = useRef<HTMLInputElement>(null)
   useEffect(() => input.current?.focus({ preventScroll: true }), [input])
 
   const [options, setOptions] = useState<ElementRef<typeof TokenOptions> | null>(null)
 
-  return (
-    <>
-      <Header title={<Trans>Select a token</Trans>} />
-      <Column gap={0.75}>
-        <Row pad={0.75} grow>
-          <ThemedText.Body1>
-            <SearchInput
-              value={query}
-              onChange={setQuery}
-              placeholder={t`Search by token name or address`}
-              onKeyDown={options?.onKeyDown}
-              onBlur={options?.blur}
-              ref={input}
-            />
-          </ThemedText.Body1>
-        </Row>
-        {Boolean(baseTokens.length) && (
-          <Row pad={0.75} gap={0.25} justify="flex-start" flex>
-            {baseTokens.map((token) => (
-              <TokenBase value={token} onClick={onSelect} key={currencyId(token)} />
-            ))}
+  const { chainId } = useActiveWeb3React()
+  const list = useTokenList()
```

Comment:
Remove `useQueryCurrencies` from `useTokenList/index.ts`, and instead re-export `useQueryTokens` from `useTokenList/querying.ts` (`export { useQueryTokens } from './querying'`), which takes the token list as a parameter. This allows re-use of the same token list for the "no tokens" check that you use in the "query" filter, and will be more performant.

Alternatively, instead of re-exporting, you can move `useQueryTokens` to its own file, maybe `useQueryTokenList.ts` (with a rename)?